apps/dashboard/app/apps/nav_bar.rb

```diff
@@ -0,0 +1,103 @@
+class NavBar
+
+  attr_reader :nav_items
+
+  def self.from_config(nav_bar)
+    nav_items = nav_bar.map do |nav_item|
+      NavBar.parse_nav_item(nav_item)
+    end
+
+    NavBar.new(nav_items: nav_items)
+  end
+
+  def initialize(nav_items: [])
+    @nav_items = nav_items
+  end
+
+  def has_items?
+    nav_items.size > 0
+  end
+
+  def to_s
+    "nav_items: #{nav_items}"
+  end
+
+  private
+
+  def self.parse_nav_item(nav_item)
+    type = nav_item.fetch(:type, nil) || NavBar.infer_type(nav_item)
+    NavItem.new(type.to_sym, nav_item)
+  end
+
+  def self.infer_type(nav_item)
+    links = nav_item.fetch(:links, nil)
+    return :nav_menu if links
+
+    url = nav_item.fetch(:url, nil)
+    return :nav_link if url
+
+    app = nav_item.fetch(:app, nil)
+    return :nav_app if app
+
+    tokens = nav_item.fetch(:tokens, nil)
+    return :nav_app if tokens
+
+    template = nav_item.fetch(:template, nil)
+    return :nav_template if template
+
+    #default type is :nav_divider
+    return :nav_divider
+  end
+
+  class NavItem
+    attr_reader :type, :items, :links, :title, :url, :icon, :new_tab, :template
+
+    def initialize(type, config)
+      config = config.to_h.compact.symbolize_keys
+
+      @type      = type
+      @items     = config.fetch(:links, []).map{|link_item| NavBar.parse_nav_item(link_item)}
+
+      @title     = config.fetch(:title, nil)
+      @url       = config.fetch(:url, nil)
+      @icon      = URI(config.fetch(:icon, "fas://cog").to_s)
+      @new_tab   = !!config.fetch(:new_tab, true)
+
+      @app       = config.fetch(:app, nil)
+      @tokens    = config.fetch(:tokens, nil)
+      @template  =  File.join("layouts/nav", config.fetch(:template, default_template))
+
+      @title = I18n.t(@title.delete_prefix("i18n://")) if @title && @title.start_with?("i18n://")
```

Comment:
This is an idea to support translations for custom titles. As mentioned in the issue, not sure how useful this might be.

Reply:
Yea I don't think this feature is wanted. You'd end up having to configure it in 2 or more places. Internationalization is about changing text you would otherwise not be able to change. But with this - you're able to put what you like directly in the YAML, so you can directly put Chinese or any language directly in the config.

I guess I'm not sure what problem it solves because the admin configuring the file can already put whatever they like here.

Reply:
OK - Will remove